components/navigation/navChild.js

```diff
@@ -65,24 +65,53 @@ const NavChild = ({ slug, page, color, className }) => {
   }
 
   let link;
+  let divider;
   let icon;
   let target;
-  let url = page.url;
+  let url;
+  if (typeof page.url !== "undefined") {
+    url = page.url;
+  } else {
+    url = "";
+  }
 
-  const isLocalPage = page.url.startsWith("/");
+  const isLocalPage = url.startsWith("/");
+  const isCrossLinkedPage = url.startsWith("https://docs.streamlit.io");
+  const isDivider = url === "";
 
-  if (!isLocalPage) {
+  if (!isLocalPage && !isCrossLinkedPage && !isDivider) {
     icon = <i className={styles.ExternalIcon}>open_in_new</i>;
     target = "_blank";
   }
 
-  if (page.isVersioned && version && isLocalPage) {
+  if (isCrossLinkedPage) {
+    url = url.replace("https://docs.streamlit.io", "");
+    icon = <i className={styles.CrossLinkedIcon}>link</i>;
+  }
+
+  if (page.isVersioned && version && isLocalPage | isCrossLinkedPage) {
     // We need to version this URL, check if the URL has a version for this version
-    const newSlug = page.url.split("/");
+    const newSlug = url.split("/");
     newSlug[0] = version;
     url = `/${newSlug.join("/")}`;
   }
 
+  if (page.name == "---") {
```

Comment:
I'm probably missing something here, but if we have this check here, why do we need all the `isDivider` logic above?

Reply:
The divider may be labeled or unlabeled. I could logic gate this a little better, but the code basically computes the divider version and category version of each element, then chooses which to return at the end. When computing the divider version, it's labeled or unlabeled which is what this portion is doing.

The reason I did it this way was to preserve the variable names, I could either dispense with the variable divider and directly put the contents in link under three conditions:
1. isDIvider and page.name == "---"
2. else isDivider
3. else

I thought having the divider variable made it a little clearer what was happening. If I change from "compute divider and link then choose," would you prefer I rename link to navItem or something?